src/emulator/storage/metadata.ts

```diff
@@ -473,3 +480,10 @@ function generateMd5Hash(bytes: Buffer): string {
   hash.update(bytes);
   return hash.digest("base64");
 }
+
+function generateETag(bytes: Buffer): string {
+  const hash = crypto.createHash("sha1");
+  hash.update(bytes);
+  // Trim padding
+  return hash.digest("base64").slice(0, -1);
+}
```

Comment:
Not really sure how GCS calculates these, but I guess it doesn't matter as long as its consistent for the emulator?

For now I've based it off the [`etag` package](https://github.com/jshttp/etag/blob/4664b6e53c85a56521076f9c5004dd9626ae10c8/index.js#L46-L50).

Reply:
I can use a combination of those 2 things, but as the [`generation` is based on the current timestamp](https://github.com/bookcreator/firebase-tools/blob/0191163aa697f2f7cc94c063841688d3c961a90a/src/emulator/storage/metadata.ts#L59), it might not be always be unique in the emulator context - it's a small possibility but could happen.

What about a combination of `name`, `generation`, `metageneration` and file contents?

Reply:
Ah sorry I think I misread unique across it objects in GCS as unique across GCS 😅
Combo of `generation` and `metageneration` sounds good 👍

Reply:
Done - etag now based on `generation/metadatageneration`